plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -178,7 +185,8 @@ public ThriftHiveMetastore(
             boolean assumeCanonicalPartitionKeys,
             boolean useSparkTableStatisticsFallback,
             ThriftMetastoreStats stats,
-            ExecutorService writeStatisticsExecutor)
+            ExecutorService writeStatisticsExecutor,
```

Comment:
for every user/identity there is a pool of Metastore client?

there is a Delegation Token cache as well, which is maintained separately.

can there be an issue, where DT is expired but the Metastore client is cached and used.